vivisect/impemu/platarch/i386.py

```diff
@@ -12,4 +17,52 @@ class i386WorkspaceEmulator(v_i_emulator.WorkspaceEmulator, e_i386.IntelEmulator
     def __init__(self, vw, logwrite=False, logread=False):
         e_i386.IntelEmulator.__init__(self)
         v_i_emulator.WorkspaceEmulator.__init__(self, vw, logwrite=logwrite, logread=logread)
-        self.setEmuOpt('i386:reponce',True)
+        self.setEmuOpt('i386:reponce', True)
+
+    def getRegister(self, index):
```

Comment:
So I actually ripped this particular section right from the AMD64 platarch which does the exact same isRegUse stuff. The getRegister stuff I modified since our taint values are a bit wonkier on i386 than amd64.

But yea, it does belong right in here. We need to be in a place where we can both call _useVirtAddr (to track register usages) and also get the register values. We can't call _useVirtAddr directly in the envi/arch/i386/emu.py emulator class since that breaks vdb/vtrace. And it feels like a full API breach to do it in the WorkspaceEmulator class (since it is an arch-specific thing).

Reply:
is it tho?  an arch-specific thing?
aren't we doing pretty basic meta-register things?  meta-registers are architecture-independent, so that's what i was referring to.

Reply:
The meta register stuff? That's fair, that's probably not particularly arch-dependent.

I guess I'm letting my biases show in that since WorkspaceEmulator doesn't strictly inherit from RegisterContext, I don't want to have it call the RegisterContext APIs. It's more personal preference since it makes it easier to navigate around.

Though if we did have the workspace emulator override getRegister, we'd probably want to formalize the isRegUse/filterRegUse/<name> function so that each architecture can define it's own ways to drop register uses that don't matter.

Reply:
table it for later.